scope_registry.go

```diff
@@ -88,28 +120,33 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 	}
 
 	buf := keyForPrefixedStringMapsAsKey(make([]byte, 0, 256), prefix, parent.tags, tags)
-	r.mu.RLock()
+	h := maphash.Hash{}
+	h.SetSeed(r.seed)
+	_, _ = h.Write(buf)
+	subscopeBucket := r.subscopes[h.Sum64()%uint64(len(r.subscopes))]
```

Comment:
How much latency does this add (for serial calls)?

Reply:
Not very clear about this. In my summary I have added the benchmarks for the non-parallel versions.

You can see that:
- When cached, this new version shows an increase of ~60 ns.
- When not cached it shows an improvement of ~300 ns. So I would like to say it is negligible.

Reply:
I think we might even be able to do a bit better - 60ns is a pretty hard hit (I understand that the gain is notable in the parallel case, but we can't add a ton of serial latency either).

I'm mostly afk today, but I'll run some tests tonight/this weekend to see if we can get it closer to parity with the non-striped version for serial cached calls.

Part of the issue is that we need to account for various modes of updating metrics, like `scope.Counter("foo").Inc(1)` calls in a relatively tight loop. Granted users should be optimizing these themselves anyway, but they often don't, so we don't want to degrade very much compared to before this change.

Reply:
Actually, I reran multiple times the benchmarks and I'm not able to get to below 700ns anymore on the old version. Both versions now sit at ~750ns.

Reply:
The benchmark shows 0.81% spent on maphash out of 31.6% (total for this benchmark).

<img width="589" alt="Screen Shot 2022-10-14 at 2 29 09 PM" src="https://user-images.githubusercontent.com/3793727/195947020-19366503-1961-48c3-9ed0-6f16e713dd92.png">

Reply:
One thing to do to address this is to allow users to disable sharding. If the application only accesses metrics serially it makes sense to do that, but I would assume most applications access them in parallel.